src/ImageSharp/Compression/Zlib/ZlibInflateStream.cs

```diff
@@ -161,29 +161,25 @@ public override int Read(byte[] buffer, int offset, int count)
             bytesToRead = Math.Min(count - totalBytesRead, this.currentDataRemaining);
             this.currentDataRemaining -= bytesToRead;
             bytesRead = this.innerStream.Read(buffer, offset, bytesToRead);
+            if (bytesRead == 0)
+            {
+                return totalBytesRead;
+            }
+
             totalBytesRead += bytesRead;
         }
 
         return totalBytesRead;
     }
```

Comment:
Since we are already touching this class: we should check the return value of `this.innerStream.Read` in `InitializeInflateStream` when it's skipping 4 bytes.

Reply:
Pushed the change together with another check.

Reply:
Good changes. Thanks!